### PR TITLE
Feat/iprod 93 Use new mcm-client to get OIDC token from hub IaM provider (Keycloak)

### PR DESCRIPTION
**Pull request by geka-evk**
ticket: https://infitx-technologies.atlassian.net/browse/IPROD-93

- Added new env vars: 
  `HUB_IAM_PROVIDER_URL` - url (with a protocol) of hub IaM provider (Keyclok)
  `AUTH_CLIENT_ID` - dfsp client ID (in Keycloak)
  `AUTH_CLIENT_SECRET` - dfsp client secret (in Keycloak)

- Removed env vars:  `AUTH_USER`  and `AUTH_PASS`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -34,7 +33,7 @@ LABEL org.label-schema.vcs-url=$VCS_URL
 LABEL org.label-schema.vcs-ref=$VCS_REF
 LABEL org.label-schema.version=$VERSION
 
-COPY tsconfig.json tsconfig-paths.js package.json package-lock.json* /opt/app/
+COPY tsconfig.json tsconfig-paths.js package*.json /opt/app/
```
Comment (aaronreynoza) [inline Dockerfile:36]: Most Docker files/node dependencies/ci pipelines,etc are update in our NodeJS versions upgrade Epic. Wonder why MCM wasn't added there. May need to check.

Comment (geka-evk) [reply Dockerfile:36]: Good point!
to be honest, IDK )
but i'll check it with Paul, and we could do it in a separate task

Comment (PaulGregoryBaker) [reply Dockerfile:36]: MCM is not part of mojaloop at the moment.

File: package.json [hunk 2]
```diff
@@ -33,8 +33,8 @@
     "@babel/eslint-parser": "^7.18.2",
     "@kubernetes/client-node": "^0.16.2",
     "@mojaloop/api-snippets": "^17.0.3",
-    "@mojaloop/sdk-standard-components": "^17.0.3",
-    "@pm4ml/mcm-client": "^2.0.11",
+    "@mojaloop/sdk-standard-components": "^17.1.3",
+    "@pm4ml/mcm-client": "github:pm4ml/mcm-client#feat/IPROD-93",
```
Comment (geka-evk) [inline package.json:37]: Will be updated to `"@pm4ml/mcm-client": "^3.0.0",` after merging **mcm-client** to _master_ branch

File: src/UIAPIServer/handlers.ts [hunk 3]
```diff
@@ -228,6 +239,9 @@ export const createHandlers = () => ({
     post: createDFSPCA,
     put: setDFSPCA,
   },
+  '/dfsp/endpoints': {
```
Comment (mdebarros) [inline src/UIAPIServer/handlers.ts:242]: Why do we need this operation?

Comment (geka-evk) [reply src/UIAPIServer/handlers.ts:242]: When I started testing all my changes locally, I saw lots of errors on _pm4ml_ ui. And I started fixing them.  But then I realized that there were to many similar errors, and i asked about it: https://infitx.slack.com/archives/C04HG5L8LDQ/p1696503562645689
So this fix was the first I made, but for now, seems, we can avoid it

**Review by mdebarros — DISMISSED** [review]
+1

**Review by vijayg10 — APPROVED** [review]

**Pull request merged**